bleak/backends/bluezdbus/manager.py

```diff
@@ -946,36 +951,38 @@ def _parse_msg(self, message: Message):
 
                 if interface == defs.DEVICE_INTERFACE:
                     # handle advertisement watchers
+                    device_path = message_path
 
                     self._run_advertisement_callbacks(
-                        message.path, cast(Device1, self_interface), changed.keys()
+                        device_path, cast(Device1, self_interface), changed.keys()
                     )
 
                     # handle device condition watchers
-                    for condition_callback in self._condition_callbacks.get(
-                        message.path, ()
-                    ):
-                        condition_callback()
+                    condition_callbacks = self._condition_callbacks.get(device_path)
+                    if condition_callbacks:
+                        for condition_callback in condition_callbacks:
+                            condition_callback(unpacked_changed)
```

Comment:
This does not take into account that properties can be invalidated (removed). IMHO, it was simpler the way it was to let the handler just get the property instead of trying to recreate the logic twice.

Reply:
made it named tuple in 5e40e914d0a20597fcf7e89df12f1457aeb8c6a4